docs/src/main/paradox/amqp.md

```diff
@@ -12,6 +12,19 @@ The AMQP connector provides Akka Stream sources and sinks to connect to AMQP ser
 
 ## Usage
 
+### Connecting to AMQP server
+
+All the AMQP connectors are configured using a @scaladoc[AmqpConnectionProvider](akka.stream.alpakka.amqp.AmqpConnectionProvider) and a list of @scaladoc[Declaration](akka.stream.alpakka.amqp.Declaration)
+
+There are two types of @scaladoc[AmqpConnectionProvider](akka.stream.alpakka.amqp.AmqpConnectionProvider):
+* @scaladoc[DefaultAmqpConnectionProvider](akka.stream.alpakka.amqp.DefaultAmqpConnectionProvider) which create a new connection for each connector and releases it when the stage stops.
+* @scaladoc[ReusableAmqpConnectionProvider](akka.stream.alpakka.amqp.ReusableAmqpConnectionProvider) which reuses the same connections for all stages. By default it closes the connection whenever the last stage using the provider stops. Optionally, it takes `automaticRelease` boolean parameter so the connection is not automatically release and the user have to release it explicitly.
```

Comment:
How could the user release it explicitly when `release` checks the boolean flag?

Reply:
Release never check a boolean flag any more since that was specific for the cached connection and we needed a generic release method.
The reusable connection offers a client counter so you can do `provider.clients.forEach(_ => provider.release)` or something like that